lib/trino-parquet/src/main/java/io/trino/parquet/writer/valuewriter/BinaryShortDecimalValueWriter.java

```diff
@@ -0,0 +1,64 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.parquet.writer.valuewriter;
+
+import io.trino.spi.block.Block;
+import io.trino.spi.type.DecimalType;
+import io.trino.spi.type.Type;
+import org.apache.parquet.column.values.ValuesWriter;
+import org.apache.parquet.io.api.Binary;
+import org.apache.parquet.schema.PrimitiveType;
+
+import static com.google.common.base.Preconditions.checkArgument;
+import static java.util.Objects.requireNonNull;
+
+public class BinaryShortDecimalValueWriter
+        extends PrimitiveValueWriter
+{
+    private final DecimalType decimalType;
+
+    public BinaryShortDecimalValueWriter(ValuesWriter valuesWriter, Type type, PrimitiveType parquetType)
+    {
+        super(parquetType, valuesWriter);
+        this.decimalType = (DecimalType) requireNonNull(type, "type is null");
+        checkArgument(this.decimalType.isShort(), "type is not a short decimal");
+        checkArgument(parquetType.getTypeLength() <= Long.BYTES, "Type length %s exceeded 8 bytes", parquetType.getTypeLength());
+    }
+
+    @Override
+    public void write(Block block)
+    {
+        for (int i = 0; i < block.getPositionCount(); i++) {
+            if (!block.isNull(i)) {
+                long value = decimalType.getLong(block, i);
+                Binary binary = Binary.fromConstantByteArray(paddingLong(value));
+                getValueWriter().writeBytes(binary);
+                getStatistics().updateStats(binary);
+            }
+        }
+    }
+
+    private byte[] paddingLong(long unscaledValue)
+    {
+        int numBytes = getTypeLength();
+        byte[] result = new byte[numBytes];
+        int shift = 8 * (numBytes - 1);
+
+        for (int i = 0; i < numBytes; i++) {
+            result[i] = (byte) (unscaledValue >> shift);
+            shift -= 8;
+        }
```

Comment:
Maybe unfold the loop with a switch on numBytes

https://github.com/trinodb/trino/blob/55df60d3bad33dbdc996e6ced9eff61f6ff9257b/lib/trino-orc/src/main/java/io/trino/orc/stream/BooleanInputStream.java#L193

Reply:
@findepi what is the gain of doing this change?

Reply:
i expect them to perform differently.